crates/bevy_ecs/src/lib.rs

```diff
@@ -1,27 +1,1045 @@
+        assert_eq!(
+            e.get::<&'static str>(),
+            Some(&"a"),
+            "&'static str is not in the removed bundle, so it should exist"
+        );
+        assert_eq!(
+            e.get::<i32>(),
+            None,
+            "i32 is in the removed bundle, so should not exist"
+        );
+        assert_eq!(
+            e.get::<f64>(),
+            None,
+            "f64 is in the removed bundle, so should not exist"
+        );
+        assert_eq!(
+            e.get::<usize>(),
+            None,
+            "usize is in the removed bundle, so should not exist"
+        );
+    }
+
+    #[test]
+    fn non_send_resource() {
+        let mut world = World::default();
+        world.insert_non_send(123i32);
+        world.insert_non_send(456i64);
+        assert_eq!(*world.get_non_send_resource::<i32>().unwrap(), 123);
+        assert_eq!(*world.get_non_send_resource_mut::<i64>().unwrap(), 456);
+    }
+
+    #[test]
+    #[should_panic]
+    fn non_send_resource_panic() {
+        let mut world = World::default();
+        world.insert_non_send(0i32);
+        std::thread::spawn(move || {
+            let _ = world.get_non_send_resource_mut::<i32>();
+        })
+        .join()
+        .unwrap();
+    }
+
+    #[test]
+    fn flags_query() {
+        let mut world = World::default();
+        let e1 = world.spawn().insert_bundle((A(0), B(0))).id();
+        world.spawn().insert(B(0));
+
+        let mut flags_query = world.query::<Option<Flags<A>>>();
+        let flags = flags_query.iter(&world).collect::<Vec<_>>();
+        let a_flags = flags[0].as_ref().unwrap();
+        assert!(flags[1].is_none());
+        assert!(a_flags.added());
+        assert!(!a_flags.mutated());
+        assert!(a_flags.changed());
+        world.clear_trackers();
+        let flags = flags_query.iter(&world).collect::<Vec<_>>();
+        let a_flags = flags[0].as_ref().unwrap();
+        assert!(!a_flags.added());
+        assert!(!a_flags.mutated());
+        assert!(!a_flags.changed());
+        *world.get_mut(e1).unwrap() = A(1);
+        let flags = flags_query.iter(&world).collect::<Vec<_>>();
+        let a_flags = flags[0].as_ref().unwrap();
+        assert!(!a_flags.added());
+        assert!(a_flags.mutated());
+        assert!(a_flags.changed());
+    }
+
+    #[test]
+    fn exact_size_query() {
+        let mut world = World::default();
+        world.spawn().insert_bundle((A(0), B(0)));
+        world.spawn().insert_bundle((A(0), B(0)));
+        world.spawn().insert_bundle((A(0), B(0), C));
+        world.spawn().insert(C);
+
+        let mut query = world.query::<(&A, &B)>();
+        assert_eq!(query.iter(&world).len(), 3);
+    }
+
+    #[test]
+    #[should_panic]
+    fn duplicate_components_panic() {
+        let mut world = World::new();
+        world.spawn().insert_bundle((1, 2));
+    }
+
+    // TODO: uncomment these tests when features are implemented
```

Comment:
Gentle reminder.

Reply:
these features likely won't be implemented this pr as they aren't explicitly needed

Reply:
i guess they aren't really encoding anything novel/useful. we can just create new tests when they are implemented.